### PR TITLE
persist extension installation state for the duration of the window's life

### LINKED ISSUE #1590: Extensions page state reset when page reopened
If the extensions page is closed and opened while extension install is being installed, the state is lost, e.g. the installing animation on the disabled button is not displayed.

### PR DESCRIPTION
**Pull request by Nokel81**
fixes #1590

known limitation: I didn't verify if an installation continues when the lens window is closed but I don't believe so since it seems to be done completely within the UI. If this isn't the case then this probably needs a longer lasting solution.

Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

**Review by panuhorsmalahti — APPROVED** [review]

File: src/renderer/components/+extensions/extensions.tsx [hunk 1]
```diff
@@ -39,25 +40,20 @@ interface InstallRequestValidated extends InstallRequestPreloaded {
...
-  private installPathValidator: InputValidator = {
+  private static installPathValidator: InputValidator = {
     message: <Trans>Invalid URL or absolute path</Trans>,
     validate(value: string) {
       return InputValidators.isUrl.validate(value) || InputValidators.isPath.validate(value);
     }
   };
 
-  @observable
-  extensionState = observable.map<string, ExtensionState>();
```
Comment (panuhorsmalahti) [inline src/renderer/components/+extensions/extensions.tsx:60]: I wonder if we could alternatively make this property `static` and thus singleton?

**Review by aleksfront — APPROVED** [review]
Looking good. Just minor thing: probably need to change store filename from `extension-install.store.ts`  to `extension-state.store.ts`.

File: src/renderer/components/+extensions/extensions.tsx (hunk 1 above)
Comment (Nokel81) [reply src/renderer/components/+extensions/extensions.tsx:60]: Probably, that is basically the same thing. I just tried to follow the same style as before.

**Comment by Nokel81** [comment]
@panuhorsmalahti Can you give some guidance about the test that is failing? I cannot see how this test was ever passing.

**Pull request merged**